scipy/_lib/tests/test__util.py

```diff
@@ -85,37 +85,39 @@ def test_serial(self):
             p = MapWrapper(0)
 
     def test_parallel(self):
-        with MapWrapper(2) as p:
-            out = p(np.sin, self.input)
-            assert_equal(list(out), self.output)
-
-            assert_(p._own_pool is True)
-            assert_(isinstance(p.pool, PWL))
-            assert_(p._mapfunc is not None)
-
-        # the context manager should've closed the internal pool
-        # check that it has by asking it to calculate again.
-        with assert_raises(Exception) as excinfo:
-            p(np.sin, self.input)
-
-        # on py27 an AssertionError is raised, on >py27 it's a ValueError
-        err_type = excinfo.type
-        assert_((err_type is ValueError) or (err_type is AssertionError))
-
-        # can also set a PoolWrapper up with a map-like callable instance
-        try:
-            p = Pool(2)
-            q = MapWrapper(p.map)
-
-            assert_(q._own_pool is False)
-            q.close()
-
-            # closing the PoolWrapper shouldn't close the internal pool
-            # because it didn't create it
-            out = p.map(np.sin, self.input)
-            assert_equal(list(out), self.output)
-        finally:
-            p.close()
+        # This is needed for Windows to get out of infinite recursion
```

Comment:
Not just Windows.  The problem also occurs on Mac OSX with Python 3.8.

Reply:
Indeed I'll fix it once I see the green tick and merge right away.